03_repair_eeg_artifacts.py

```diff
@@ -0,0 +1,141 @@
+# get EOG-channel names
+eogs = raw.copy().pick_types(eog=True).ch_names
+
+for eog in eogs:
+    eog_epochs = create_eog_epochs(raw,
+                                   ch_name=eog,
+                                   reject_by_annotation=True)
+
+    # create average blink
+    eog_evoked = eog_epochs.average()
+    eog_evoked.apply_baseline(baseline=(None, -0.2))
+
+    # find components that correlate with activity recorded at eog
+    # channel in question
+    eog_indices, eog_scores = ica.find_bads_eog(eog_epochs,
+                                                ch_name=eog,
+                                                reject_by_annotation=True)
+
+    # if any "bad" components found:
+    if eog_indices and any(eog_indices) not in ica.exclude:
+
+        for eog_i in eog_indices:
+            # add component to list for exclusion
+            ica.exclude.append(eog_i)  # noqa
+
+            # create summary plots
+            fig = ica.plot_properties(eog_epochs,
+                                      picks=eog_i,
+                                      psd_args={'fmax': 35.},
+                                      image_args={'sigma': 1.},
+                                      show=False)[0]
+            plt.close(fig)
+            fig_evoked = ica.plot_sources(eog_evoked, show=False)
+            plt.close(fig_evoked)
+
+            # create HTML report
+            with open_report(fname.report(subject=subject)[0]) as report:
+                report.add_figs_to_section(fig, 'Bad components identified '
+                                                'by %s electrode' % eog,
+                                           section='ICA',
+                                           replace=True)
+                report.add_figs_to_section(fig_evoked, 'Components sources as '
+                                                       'identified '
+                                                       'by %s electrode' % eog,
+                                           section='ICA',
+                                           replace=True)
+                report.save(fname.report(subject=subject)[1], overwrite=True,
+                            open_browser=False)
+
+# 4) Find any further components via correlation with template ICA
+# (just in case previous step missed any bad components)
+
+# load template file
+template_raw_file = fname.output(subject=subject,
+                                 processing_step='artefact_detection',
+                                 file_type='raw.fif')
+template_raw = read_raw_fif(template_raw_file)
+
+# and template ICA
+template_ica_file = fname.output(subject=subject,
+                                 processing_step='fit_ica',
+                                 file_type='ica.fif')
+template_ica_file = read_ica(template_ica_file)
+
+# compute correlations with template
+corrmap(icas=[template_ica_file, ica],
+        template=(0, 0), threshold=0.9, label='blink_up', plot=False)
```

Comment:
also here:
in `template=(0, 0)` 
the first 0 in the tuple refers to the 0th ica in the icas argument (which would be the template ICA), the second 0 refers to the 0th component of that ica solution (usually a strong component that tracks eye movements for instance).
not saying the `(0, 0)`ist wrong, but definetily double check the numbers and see if you are correlating with meaningful components

Reply:
And check if they match the `label` argument